sympy/sets/fancysets.py

```diff
@@ -1433,6 +1431,11 @@ def _intersect(self, other):
 
     def _union(self, other):
 
+        if other.is_subset(S.Reals):
+
+            # treat a subset of reals as a complex region
+            other = ComplexRegion(other * FiniteSet(0), polar=self.polar)
```

Comment:
`other*FiniteSet(0)` is a representation in rectangular coordinates. Hence polar should not be True.